extensions/ql-vscode/src/remote-queries/variant-analysis-processor.ts

```diff
@@ -158,12 +158,13 @@ function processApiRepoStatus(analysisStatus: ApiVariantAnalysisRepoStatus): Var
   }
 }
 
-function processApiStatus(status: ApiVariantAnalysisStatus): VariantAnalysisStatus {
-  switch (status) {
-    case 'in_progress':
-      return VariantAnalysisStatus.InProgress;
-    case 'completed':
-      return VariantAnalysisStatus.Succeeded;
+function processApiStatus(status: ApiVariantAnalysisStatus, failureReason: string | undefined): VariantAnalysisStatus {
+  if (status === 'in_progress') {
+    return VariantAnalysisStatus.InProgress;
+  } else if (failureReason !== undefined) {
```

Comment:
Is this different than `if (failureReason)` ?

Reply:
It is technically different because the empty string is falsy. Practically I'm not sure it'll make a difference.

If the failure message is set to the empty string that's not great, but I'd say it's a bug earlier in the system rather than here. So here I generally prefer to be explicit about checking for undefined rather than using the implicit conversion to a boolean.